pkgs/by-name/gv/gvmd/package.nix

```diff
@@ -0,0 +1,114 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchurl
+, cmake
+, gvm-libs
+, pkg-config
+, gnutls
+, xml-tooling-c
+, glib
+, doxygen
+, git
+, libbsd
+, libical
+, libpkgconf
+, gpgme
+, mount
+, libidn2
+, graphviz
+, icu
+, p11-kit
+, pcre2
+, libxslt
+, libtasn1
+, postgresql
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gvmd";
+  version = "23.4.0";
+
+  src = fetchFromGitHub {
+    owner = "greenbone";
+    repo = "gvmd";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-Bxq6o98viMMwt2eoiZgPDTYqD/z6jRWYLgH/ZJvkpV4=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    gvm-libs
+    graphviz
+    mount
+    gnutls
+    glib
+    git
+    doxygen
+    pcre2
+    p11-kit
+    libbsd
+    libidn2
+    libtasn1
+    icu
+    libical
+    xml-tooling-c
+    gpgme
+    libxslt
+    postgresql
+  ];
+
+  buildPhase = ''
+    substituteInPlace $src/src/sql_pg.c --replace "#include postgresql/libpq-fe.h" "#include ${postgresql}/include/libpq-fe.h"
+    mkdir -p $out/{bin,etc,sbin,lib,usr/{share,include},run/gvmd/gvmd.pid}
+    mkdir -p \
+      $out/var/log/gvm \
+      $out/var/lib/openvas/plugins \
+      $out/var/lib/gvm/cert-data \
+      $out/var/lib/gvm/users \
+      $out/var/lib/gvm/scap-data \
+      $out/var/lib/gvm/CA \
+      $out/var/lib/gvm/private/CA \
+      $out/var/lib/gvm/data-objects \
+      $out/var/lib/gvm/feed-update.lock
+    cmake -DSYSCONFDIR=$out/etc \
```

Comment:
It's better to split into `installPhase`? Also can `cmakeFlags` handle this?

Reply:
split into postBuildPhase to reduce complexity.

Reply:
cmakeFlags does not really produce the result compared to using the buildPhase \ cmake ... reason been that a gvm-checking is required to get checking lock. buildPhase does not produce such error.

Reply:
> cmakeFlags does not really produce the result compared to using the buildPhase \ cmake ... reason been that a gvm-checking is required to get checking lock. buildPhase does not produce such error.

Could you explain that in detail?